### DISCUSSION TITLE
Is there any way to make new block from entire page's block?

### DISCUSSION THREAD

**Discussion by ehsunoo** [GrapesJS/grapesjs#5756] [Q&A]
I want to create a new block using the design from other blocks such as input, select, etc. and save the entire page (form) template in the database. 
I need to retrieve all existing blocks on the editor canvas as an array and send them as a request to the server. 
I have checked the APIs but I didn't find a suitable one for getting all existing blocks on the editor canvas. 
It seems that using APIs like "getProjectData" is not suitable for my goal. 
Am I correct? 
Or is there any other solution for this?

**Comment by artf** [comment]
You can save the component JSON and reuse it later
```js
const component = editor.getSelected();
const cmpJSON = JSON.parse(JSON.stringify(component));
// ...
editor.getSelected().append(cmpJSON);
```

**Reply by ehsunoo** [reply]
Thanks Artur,
I checked that,
but it doesn't copy the component's styles.
Is there any solution?

I want to copy all components with their attributes and save them to my DB, then fetch them from DB to create a new block using editor.Blocks.add().
